### PR TITLE
[docs] remove duplicated sources-geojson section

### LINKED ISSUE #7873: Wrong StyleSpec Docs for Geojson Source
Not sure if this is the right repo to report bug about the mapbox gl js doc, but doing anyway.
(StyleSpec Docs for Geojson Source)[https://docs.mapbox.com/mapbox-gl-js/style-spec/#sources-geojson]

There are two sections of geojson source in the doc. The first geojson source doc list properties like 
`tiles`/`scheme`/`url` (properties from TileJSON). And I guess the first one is wrong.

**mapbox-gl-js version**:

**browser**:

### Steps to Trigger Behavior

 1.
 2.
 3.

### Link to Demonstration

https://jsbin.com/

### PR DESCRIPTION
**Pull request by katydecorah**
The sources-geojson found itself duplicated and nested inside the vector-sources section. This PR removes the duplicated section.

Fixes #7873

### CODE REVIEW

File: docs/pages/style-spec.js [hunk 1]
```diff
@@ -539,11 +481,16 @@ export default class extends React.Component {
                                         android: '4.2.0',
                                         ios: '3.4.0',
                                         macos: '0.3.0'
+                                    },
+                                    'line distance metrics': {
```
Comment (colleenmcginnis) [inline docs/pages/style-spec.js]: Where is this coming from @katydecorah? Was this addition intentional?

Comment (katydecorah) [reply docs/pages/style-spec.js]: Yes, I pulled it from the duplicated content and checked the git history to make sure `line distance metrics` should be included here.
